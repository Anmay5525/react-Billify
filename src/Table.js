const Table = ({ fields, data }) => {
  return (
    <div>
      <table className="table">
        <thead>
          {fields.map((el) => {
            return <th>{el.toUpperCase().replace("_", " ")}</th>;
          })}
        </thead>
        {data.map((el) => {
          return (
            <tr class="tr">
              {fields.map((el1) => {
                return <td>{el[el1]}</td>;
              })}
            </tr>
          );
        })}
      </table>
    </div>
  );
};

export default Table;
