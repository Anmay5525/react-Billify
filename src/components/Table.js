import React from "react";
import PropTypes from "prop-types";

const Table = ({ fields, data }) => {
  return (
    <div>
      <table className="table">
        <thead>
          <tr>
            {fields.map((el) => {
              return (
                <th key={JSON.stringify(el)}>
                  {el.toUpperCase().replace("_", " ")}
                </th>
              );
            })}
          </tr>
        </thead>
        <tbody>
          {data.map((el) => {
            return (
              <tr
                className="tr"
                key={JSON.stringify(el) + window.performance.now()}
              >
                {fields.map((field) => {
                  return (
                    <td
                      key={
                        JSON.stringify(el) +
                        JSON.stringify(field) +
                        window.performance.now()
                      }
                    >
                      {el[field]}
                    </td>
                  );
                })}
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

Table.propTypes = {
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
  data: PropTypes.arrayOf(
    PropTypes.objectOf(
      PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.func,
        PropTypes.instanceOf(Date),
      ])
    )
  ).isRequired,
};

export default Table;
