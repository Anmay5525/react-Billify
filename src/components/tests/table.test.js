import React from 'react';
import Table from '../Table';
import { configure, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter:new Adapter()});


describe('<Table/>',()=>{
    it('renders correctly',()=>{
        const tree=shallow(<Table 
            fields={["name", "description", "amount", "currency"]}
            data={[{
                name: "item1Name",
                description: "description1",
                amount: 10000,
                currency: "INR"
            },{
                name: "item1Name",
                description: "description1",
                amount: 10000,
                currency: "INR"
            }]}
            />);
        expect(tree).toMatchSnapshot();
    });
})