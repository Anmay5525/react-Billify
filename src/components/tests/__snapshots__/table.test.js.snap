// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> renders correctly with matching fieldnames and data keys 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          H1
        </th>
        <th>
          H2
        </th>
        <th>
          H3
        </th>
        <th>
          H4
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="tr"
      >
        <td>
          name1
        </td>
        <td>
          description
        </td>
        <td>
          10000
        </td>
        <td>
          INR
        </td>
      </tr>
      <tr
        className="tr"
      >
        <td>
          name2
        </td>
        <td>
          description
        </td>
        <td>
          20000
        </td>
        <td>
          INR
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Table/> renders correctly with non-matching fieldnames and data keys 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          H1
        </th>
        <th>
          H2
        </th>
        <th>
          H3
        </th>
        <th>
          H4
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="tr"
      >
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr
        className="tr"
      >
        <td />
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Table/> renders correctly without data 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          H1
        </th>
        <th>
          H2
        </th>
        <th>
          H3
        </th>
        <th>
          H4
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
